Sources/ClientRuntime/Endpoints/EndpointResolverMiddleware.swift

```diff
@@ -0,0 +1,112 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import class Smithy.Context
+import struct SmithyHTTPAPI.Endpoint
+import class SmithyHTTPAPI.SdkHttpRequest
+import class SmithyHTTPAPI.SdkHttpRequestBuilder
+import struct SmithyHTTPAuthAPI.SelectedAuthScheme
+import enum SmithyHTTPAuthAPI.SigningAlgorithm
+import enum SmithyHTTPAuthAPI.SigningPropertyKeys
+
+public struct EndpointResolverMiddleware<OperationStackOutput, Params: EndpointsRequestContextProviding>: Middleware {
```

Comment:
This is a static, reusable endpoint resolver middleware.  It differs from the previously code-generated middleware in that:
- It takes a generic set of endpoint params.
- It takes a block that resolves the endpoint rather than the endpoint resolver itself.  This removes the associated type linkage between endpoint resolver & params.  The block is code generated when the endpoint resolver is created.

Reply:
This middleware is generic Smithy. There is a similar AWS-specific middleware as well.